packages/venia-concept/src/components/ProductOptions/tile.js

```diff
@@ -1,38 +1,39 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import { bool, number, oneOfType, shape, string } from 'prop-types';
 
 import classify from 'src/classify';
 import defaultClasses from './tile.css';
 
 class Tile extends Component {
     static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string
+        classes: shape({
+            root: string
         }),
-        item: PropTypes.shape({
-            id: PropTypes.string,
-            name: PropTypes.string
+        hasFocus: bool,
+        isSelected: bool,
+        item: shape({
+            id: oneOfType([number, string]),
+            label: string
         })
     };
 
     render() {
-        const { classes, item } = this.props;
-        const { name } = item;
+        const {
+            classes,
+            hasFocus, // eslint-disable-line
+            isSelected,
+            item,
```

Comment:
This shares a lot of code with `Swatch` and `SwatchList`. Should we consider a wrapper element or a List variant that has a focus concept?

Reply:
I'm curious to know the reasoning behind why these are being expanded out again. In my original pull request these were accidentally merged back in (from the `default-pragma` branch). Instead of creating separate components, there was one component (`option.js`) which would load different CSS files based on the option type.

Reply:
@zetlen They only seem to share code because they're implementations of the same interface. The shared code has already been consolidated in `List` and `Item`, in Peregrine.

@pcvonz When I [originally created](https://github.com/magento-research/pwa-studio/pull/62) `SwatchList`, `TileList`, and their respective children, the UI part of the product options story was basically complete. The rendered output of `Swatch` and `Tile` are different, so they're intentionally separate components.